### ISSUE TITLE
Scrolls over div element on bottom

### ISSUE THREAD

**Issue by Mojo90** [abouolia/sticky-sidebar#87]
Really cool plugin! The only problem for me is that in different browsers (Chrome, Safari, Firefox) the sidebar sometimes scrolls into the other div element on bottom and this is very ugly (working example https://web.staging.binando.com/features/collection-containers): 
<img width="752" alt="screenshot 2018-09-06 23 08 33" src="https://user-images.githubusercontent.com/2122156/45185304-00683500-b22a-11e8-8018-b8661474ae7b.png">

My code is:
`JS:`
```
var stickySidebar = new StickySidebar('#features_sidebar', {
          topSpacing: 40,
          bottomSpacing: 40,
          containerSelector: 'section.features .container',
          innerWrapperSelector: '.sidebar__inner',
          resizeSensor: true,
          stickyClass: 'is-affixed',
          minWidth: 0
        });
```
--> adjust `bottomSpacing` changes nothing even a 0 or a high number.

`HTML:`
```
<section class="features">
    <div class="container">
        <div class="row">
          <div id="features_sidebar" class="col-sm-2 col-sm-offset-1 sidebar">
              <div class="sidebar__inner">
                <div style="text-align:right;">
                  <ul class="nav">
                    <?php foreach($features as $index => $feature): ?>
                      <li><a data-target="#<?= $feature['unique_id']?>" href="#<?= $feature['unique_id']?>"><?= $feature['heading']?></a></li>
                    <?php endforeach; ?>
                  </ul>
                </div>
              </div>
          </div>
          <div id="features_content" class="content col-sm-9">
            <div class="content__inner">
              <!-- php code to print the content -->
            </div>
          </div>
        </div>
    </div>
</section>
```

Any hints to prevent this highly appreciated!

**Comment by 89gsc** [comment]
Same issue have another issue with this plugin as well where at the top on scrolling back up it goes over the top of the content and jumps back :/ 

Thinking of looking for another plugin I needed somthing quick but this is turning into a bit of a headache I will see if I can resolve and let you know otherwise assume I gave up. 

EDIT - More Info:

Video: https://screencast-o-matic.com/watch/cFQqFCq8Sn

Code Used:
```
if ($('.sidebar-and-content'.length)) {
      new StickySidebar('.sidebar', {
        topSpacing: 88,
        bottomSpacing: 40,
        containerSelector: '.sidebar',
        innerWrapperSelector: '.sidebar-inner',
        stickyClass: 'is-stuck',
      })

      setTimeout(function () {
        window.dispatchEvent(new Event('resize'));
      }, 500)
}
```

Used within a sage roots wordpress theme which uses bootstrap 4. 

Note: containerSelector changing to the outer outer container makes no difference to this behaviour,

**Comment by Mojo90** [comment]
Yeah quite ugly and not sure where the error is. If you have an easy working alternative feel free to let me know. Otherwise some help/comments from the maintainers would be very nice

**Comment by octojoerg** [comment]
same problem here! has anybody a solution?

**Comment by joergsteinhauer** [comment]
> *offtopic: hey octojoerg, same here. lets try with the private GH account.

I found a solution for my case, but have no idea if it can help others... but anyway, here are the detail information:

**Problem:**
Sidebar jumps by scrolling down and top

Solution: add _outer container_ with padding, remove padding from _inner container_. Pay attention to the nested elements!

**Initial HTML/CSS:**

```HTML
<header>
  <div class="inner-container">
    <main>
    <aside class="sticky-sidebar">
<footer>
```

```CSS
.inner-container {
  padding: 2em
}
```

**Fixed HTML/CSS:**

```HTML
<header>
  <div class="outer-container">
    <div class="inner-container">
      <main>
      <aside class="sticky-sidebar">
<footer>
```

```CSS
.inner-container {
  padding: 0
}
.outer-container {
  padding: 2em
}
```

**Comment by cagonzalezcs** [comment]
I encountered this issue as well and found that the problem I was having in particular was with the initialization call for the sticky sidebar being placed within a $(document).ready() callback. I found that waiting until $(window).on('load') solved the issue of the sticky element overlapping the content area at the bottommost segment.

**Comment by Mojo90** [comment]
@cagonzalezcs unfortunately that did not work for me but I am using query > 3.x so I tried `$(window).on('load', function () {` instead. But thx fur the hint

**Comment by gambhiraman** [comment]
Use this link below for detecting height change:

https://stackoverflow.com/questions/14866775/detect-document-height-change

Call your sticky sidebar init function in the callback provided by the onElementHeightChange function.
Try it once it worked for me.

**Comment by Mojo90** [comment]
It works for me but not all the time. Sometimes yes, sometimes no, really weird.
